test/session-pool.js

```diff
@@ -16,2022 +16,1575 @@
+            assert.strictEqual(sessionPool.options.maxWrites, 60);
+            assert.strictEqual(sessionPool.options.minReads, 40);
+            assert.strictEqual(sessionPool.options.minWrites, 60);
+          });
+        });
 
-      sessionPool = new SessionPool(DATABASE);
-      assert.deepEqual(sessionPool.acquireQueue_, {
-        concurrency: 1,
+        it('should override DEFAULT for minReads and minWrites', function() {
+          return sessionPool.close().then(function() {
+            sessionPool = new SessionPool(DATABASE, {
+              writes: 0.5,
+              maxReads: 40,
+              min: 20,
+            });
+            assert.strictEqual(sessionPool.options.acquireTimeout, Infinity);
+            assert.strictEqual(sessionPool.options.concurrency, 10);
+            assert.strictEqual(sessionPool.options.maxWait, 50);
+            assert.strictEqual(sessionPool.options.idlesAfter, 50);
+            assert.strictEqual(sessionPool.options.max, 100);
+            assert.strictEqual(sessionPool.options.writes, 0.5);
+            assert.strictEqual(sessionPool.options.min, 20);
+            assert.strictEqual(sessionPool.options.maxReads, 40);
+            assert.strictEqual(sessionPool.options.maxWrites, 60);
+            assert.strictEqual(sessionPool.options.minReads, 20);
+            assert.strictEqual(sessionPool.options.minWrites, 20);
+          });
+        });
+
+        it('should throw when writes is greater than 1', function() {
+          assert.throws(function() {
+            return new SessionPool(DATABASE, {writes: 50});
+          }, /Write percentage should be represented as a float between 0\.0 and 1\.0\./);
+        });
       });
-    });
 
-    it('should create a request queue', function() {
-      var poolOptions = {
-        concurrency: 11,
-      };
+      it('should set isOpen to false', function() {
+        assert.strictEqual(sessionPool.isOpen, false);
+      });
 
-      pQueueOverride = function(options) {
-        return options;
-      };
+      it('should localize database request functions', function() {
+        assert.strictEqual(sessionPool.request_, DATABASE.request);
+        assert.strictEqual(sessionPool.requestStream_, DATABASE.requestStream);
+      });
 
-      sessionPool = new SessionPool(DATABASE, poolOptions);
-      assert.deepEqual(sessionPool.requestQueue_, {
-        concurrency: poolOptions.concurrency,
+      it('should set read and write pool sizes to 0', function() {
+        assert.strictEqual(sessionPool.getStats().readPool.size, 0);
+        assert.strictEqual(sessionPool.getStats().writePool.size, 0);
       });
-    });
 
-    it('should create handles for intervals', function() {
-      assert.strictEqual(sessionPool.evictHandle_, null);
-      assert.strictEqual(sessionPool.pingHandle_, null);
-    });
+      it('read and write pools should be initialized with the parameters passed in', function() {
```

Comment:
Can we break this up into 2 separate tests, one for writes and one for reads.

Reply:
Updated